src/proto/Solana.proto

```diff
@@ -22,16 +22,33 @@ message WithdrawStake {
     uint64 value = 2;
 }
 
+message CreateTokenAccount {
+    // main account -- can be same as signer, or other main account (if done on some other account's behalf)
+    string main_address = 1;
+    string token_mint_address = 2;
+    string token_address = 3;
+}
+
+message TokenTransfer {
+    string token_mint_address = 1;
+    string sender_token_address = 2;
```

Comment:
```suggestion
    string sender_address = 2;
```

same for `recipient_token_address`

Reply:
Distinction between main address and token address is important, I'd rather keep it verbose.